src/plans/create/steps/virtual-machines/VirtualMachinesStepFooter.tsx

```diff
@@ -0,0 +1,166 @@
+import { useWatch } from 'react-hook-form';
+import { useInventoryVms } from 'src/modules/Providers/views/details/tabs/VirtualMachines/utils/hooks/useInventoryVms';
+
+import {
+  Button,
+  ButtonVariant,
+  Modal,
+  ModalVariant,
+  useWizardContext,
+} from '@patternfly/react-core';
+import { isEmpty } from '@utils/helpers';
+import { useForkliftTranslation } from '@utils/i18n';
+
+import CreatePlanWizardFooter from '../../CreatePlanWizardFooter';
+import { useCreatePlanFormContext } from '../../hooks';
+import type { ProviderVirtualMachine } from '../../types';
+import { GeneralFormFieldId } from '../general-information/constants';
+
+import { VmFormFieldId } from './constants';
+import { hasCriticalConcern } from './utils';
+import VirtualMachinesTable from './VirtualMachinesTable';
+
+const VirtualMachinesStepFooter: FC = () => {
+  const { t } = useForkliftTranslation();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [pendingVmIds, setPendingVmIds] = useState<string[]>();
+  const { goToNextStep } = useWizardContext();
+  const { control, setValue, trigger } = useCreatePlanFormContext();
+  const [sourceProvider, selectedVms] = useWatch({
+    control,
+    name: [GeneralFormFieldId.SourceProvider, VmFormFieldId.Vms],
+  });
+  const [vmInventoryData] = useInventoryVms({ provider: sourceProvider });
+
+  const vmsWithCriticalIssues = useMemo(
+    () =>
+      Object.fromEntries(
+        Object.entries(selectedVms ?? {}).filter(([_, vm]) => hasCriticalConcern(vm)),
+      ),
+    [selectedVms],
+  );
+
+  const pendingVms = useMemo(
+    () =>
+      pendingVmIds
+        ? Object.fromEntries(pendingVmIds.map((id) => [id, selectedVms[id]]))
+        : vmsWithCriticalIssues,
+    [pendingVmIds, selectedVms, vmsWithCriticalIssues],
+  );
+
+  // Get VM data for the critical concern VMs to display in modal
+  const criticalVmData = useMemo(
+    () => vmInventoryData.filter(({ vm }) => hasCriticalConcern(vm) && vm.id in selectedVms),
+    [vmInventoryData, selectedVms],
+  );
+
+  // Validate form and handle next step or show warning modal
+  const handleNextStep = useCallback(async () => {
+    const isValid = await trigger(undefined, { shouldFocus: true });
+
+    if (isValid) {
+      // Skip modal if no critical issues found
+      if (isEmpty(Object.keys(vmsWithCriticalIssues))) {
+        await goToNextStep();
+        return;
+      }
+
+      // Show warning modal for VMs with critical issues
+      setIsModalOpen(true);
+    }
+  }, [trigger, vmsWithCriticalIssues, goToNextStep]);
+
+  const closeModal = useCallback(() => {
+    setIsModalOpen(false);
+    setPendingVmIds(undefined);
+  }, []);
+
+  // Handle VM selection updates and navigation
+  const updateVmSelectionAndProceed = useCallback(
+    async (vms: Record<string, ProviderVirtualMachine>) => {
+      setValue(VmFormFieldId.Vms, vms);
+      closeModal();
+
+      if (isEmpty(vms)) {
+        await trigger(VmFormFieldId.Vms);
+      } else {
```

Comment:
no need for else u can simply return here

Reply:
What exactly is the inefficiency of just using an 'else' statement that prompts these comments?